src/App.tsx

```diff
@@ -1,223 +1,183 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React, { useContext, useEffect, useRef } from 'react';
-import { AuthContext } from './components/Auth/AuthContext';
+import React, {
+  useContext,
+  useEffect,
+  useRef,
+  memo,
+  useState,
+  useMemo,
+  useCallback,
+} from 'react';
 
-export const App: React.FC = () => {
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+import { AuthContext } from './components/Auth/AuthContext';
+import { Header } from './components/Header';
+import { TodoList } from './components/TodoList/TodoList';
+import { Footer } from './components/Footer';
+import { ErrorNotification } from './components/ErrorNotification';
+
+import { FilterType } from './types/FilterType';
+import { Todo } from './types/Todo';
+import { getTodos, deleteTodo, createTodo } from './api/todos';
+
+export const App: React.FC = memo(() => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [isError, setIsError] = useState(false);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [filterType, setFilterType] = useState(FilterType.All);
+  const [isTodoAdding, setIsAdding] = useState(false);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
   const user = useContext(AuthContext);
   const newTodoField = useRef<HTMLInputElement>(null);
 
   useEffect(() => {
-    // focus the element with `ref={newTodoField}`
+    if (user) {
+      getTodos(user.id)
+        .then(setTodos)
+        .catch(() => {
+          setIsError(true);
+          setErrorMessage('Can\'t load todos');
+        });
+    }
+  }, []);
+
+  useEffect(() => {
     if (newTodoField.current) {
       newTodoField.current.focus();
     }
   }, []);
 
+  const filteredTodos = useMemo(() => {
+    if (filterType === FilterType.All) {
+      return todos;
+    }
+
+    return todos.filter(todo => {
+      switch (filterType) {
+        case FilterType.Active:
+          return !todo.completed;
+
+        case FilterType.Completed:
+          return todo.completed;
+
+        default:
+          throw new Error('Invalid type');
+      }
```

Comment:
It's better to make  a filter inside the switch, 'cause now swich is triggered on every element of an array